charts_flutter/lib/src/cartesian_chart.dart

```diff
@@ -14,15 +14,34 @@
 // limitations under the License.
 
 import 'dart:collection' show LinkedHashMap;
-
 import 'package:meta/meta.dart' show immutable, protected;
 import 'package:nimble_charts/src/base_chart.dart' show BaseChart;
 import 'package:nimble_charts/src/base_chart_state.dart' show BaseChartState;
 import 'package:nimble_charts_common/common.dart' as common
     show AxisSpec, BaseChart, CartesianChart, NumericAxis, NumericAxisSpec;
 
+/// Base class for all cartesian charts.
+///
+/// A cartesian chart is a chart that uses a cartesian coordinate system - it
+/// has a domain axis and a measure axis.
 @immutable
 abstract class CartesianChart<D> extends BaseChart<D> {
+  /// Creates a [CartesianChart].
```

Comment:
@sanarawal7 unfortunately, this looks wrong. There are only 4 parameters for the constructor, but the doco here has for more than that.